File: coyote/models/msg_topic_configure_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgTopicConfigureIn(BaseModel):
    namespace: t.Optional[str] = None

    partitions: int


class _MsgTopicConfigureIn(BaseModel):
    namespace: t.Optional[str] = None

    topic: str

    partitions: int

File: src/models/msg_topic_configure_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize)]
pub struct MsgTopicConfigureIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub namespace: Option<String>,

    pub partitions: u16,
}

impl MsgTopicConfigureIn {
    pub fn new(partitions: u16) -> Self {
        Self {
            namespace: None,
            partitions,
        }
    }

    pub fn with_namespace(mut self, value: impl Into<Option<String>>) -> Self {
        self.namespace = value.into();
        self
    }
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub(crate) struct MsgTopicConfigureIn_ {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub namespace: Option<String>,

    pub topic: String,

    pub partitions: u16,
}
